test/parallel_api/iterator/input_data_sweep_host_iter.pass.cpp

```diff
@@ -0,0 +1,82 @@
+// -*- C++ -*-
+//===-- input_data_sweep_host_iter.pass.cpp -------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/utils.h"
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(iterator)
+
+#include "input_data_sweep.h"
+
+#include "support/utils_invoke.h"
+
+//This test is written without indirection from invoke_on_all_hetero_policies to make clear exactly which types
+// are being tested, and to limit the number of types to be within reason.
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+template <typename T, int __recurse, typename Policy>
+void
+test(Policy&& policy, T trash, size_t n, const std::string& type_text)
+{
+    if (TestUtils::has_types_support<T>(policy.queue().get_device()))
+    {
+
+        TestUtils::usm_data_transfer<sycl::usm::alloc::shared, T> copy_out(policy.queue(), n);
+        oneapi::dpl::counting_iterator<int> counting(0);
+        // host iterator
+        std::vector<T> host_iter(n);
+        wrap_recurse<__recurse, 0, /*__read =*/true, /*__reset_read=*/true, /*__write=*/true,
+                     /*__check_write=*/true, /*__usable_as_perm_map=*/true, /*__usable_as_perm_src=*/false,
+                     /*__is_reversible=*/true>(policy, host_iter.begin(), host_iter.end(), counting,
+                                               copy_out.get_data(), host_iter.begin(), copy_out.get_data(), counting,
+                                               trash, std::string("host_iterator<") + type_text + std::string(">"));
+    }
+    else
+    {
+        TestUtils::unsupported_types_notifier(policy.queue().get_device());
+    }
+}
+
+#endif //TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+
+    constexpr size_t n = 10;
+
+    auto q = TestUtils::get_test_queue();
+
+    auto policy = TestUtils::make_new_policy<class Kernel1>(q);
+
+    auto policy1 = TestUtils::create_new_policy_idx<decltype(policy), 0>(policy);
+    auto policy2 = TestUtils::create_new_policy_idx<decltype(policy), 1>(policy);
+    auto policy3 = TestUtils::create_new_policy_idx<decltype(policy), 2>(policy);
+    auto policy4 = TestUtils::create_new_policy_idx<decltype(policy), 3>(policy);
+
+    // baseline with no wrapping
```

Comment:
Yeah, my original intention was to include this as well for the reason you describe, but the test would need to be specialized for this case (to not use counting_iterator as source & verification of sink). 
I'll see if it is reasonable to add without adding too much complexity.

Reply:
I've added boolean types for host iterators.  Thanks.

Reply:
After discussion with @MikeDvorskiy , I have learned that `std::vector<bool>::iterator` for any allocator is undefined behavior as input of oneDPL due to the optimizations standard library implementations add to turn it into a bitfield.  This definitely doesn't work for host policies as the data is used directly and elements occupy an bit in a bitfield rather than a whole type.  In `icpx`, this does technically work with a device policy and `dpcpp` backend (due to the copy step into a `sycl::buffer`).  

However, since this depends on standard library implementation and policy, we determined it is best to just describe it as undefined behavior.  I've added #1443 to better document this.